src/redux/services/resource.ts

```diff
@@ -275,10 +277,16 @@ export function getNamespaces(resourceMap: ResourceMapType) {
   return namespaces;
 }
 
-export async function getTargetClusterNamespaces(kubeconfigPath: string, context: string): Promise<string[]> {
+export async function getTargetClusterNamespaces(kubeconfigPath: string, context: string, clusterAccess?: ClusterAccess[]): Promise<string[]> {
+  const hasFullAccess = clusterAccess?.some((ca) => ca.hasFullAccess);
+  const clusterAccessNamespaces = clusterAccess?.map((ca) => ca.namespace) || [];
+  if (!hasFullAccess) {
+    return clusterAccessNamespaces;
+  }
+
   try {
     const kubeClient = createKubeClient(kubeconfigPath, context);
-    const namespaces = await NamespaceHandler.listResourcesInCluster(kubeClient);
+    const namespaces = await NamespaceHandler.listResourcesInCluster(kubeClient, { namespace: '' });
```

Comment:
it's a bit weird to have to specify the `namespace` option for this, but I'm guessing it helped a lot to refactor the code and finding where it had to be added

Reply:
Yes, it is a bit weird, in theory I can mark that field as optional now